files/en-us/web/javascript/reference/global_objects/asyncgeneratorfunction/index.html

```diff
@@ -0,0 +1,100 @@
+
+<p>Note that <code>AsyncGeneratorFunction</code> is not a global object. It could be obtained by evaluating the following code.</p>
+
+<pre class="brush: js">Object.getPrototypeOf(async function*(){}).constructor
+</pre>
+
+<h2 id="Syntax">Syntax</h2>
+
+<pre  class="brush: js">new GeneratorFunction(functionBody)
+new GeneratorFunction(arg1, functionBody)
+new GeneratorFunction(arg1, arg2, functionBody)
+</pre>
+
+<h3 id="Parameters">Parameters</h3>
+
+<dl>
+  <dt><code>arg1</code>, <code>arg2</code>, ... <code>arg<em>N</em></code></dt>
+  <dd>Names to be used by the function as formal argument names. Each must be a {{jsxref('String')}} that corresponds to a valid JavaScript identifier or a list of such strings separated with a comma; for example "<code>x</code>", "<code>theValue</code>", or "<code>a,b</code>".</dd>
+  <dt><code>functionBody</code></dt>
+  <dd>A {{jsxref('String')}} containing the JavaScript statements comprising the function definition.
+  </dd>
+</dl>
+
+<h2 id="Description">Description</h2>
+
+<p>{{jsxref("Statements/async_function*", "async generator function")}} objects created with the <code>AsyncGeneratorFunction</code> constructor are parsed when the function is created. This is less efficient than declaring a generator function with an {{jsxref("Statements/async_function*", "async function* expression")}} and calling it within your code, because such functions are parsed with the rest of the code.</p>
+
+<p>All arguments passed to the function are treated as the names of the identifiers of the parameters in the function to be created, in the order in which they are passed.</p>
+
+<div class="notecard note">
+  <h3>Note:</h3>
+  <p>{{jsxref("Statements/async_generator_function*", "async generator functions")}} created with the <code>AsyncGeneratorFunction</code> constructor do not create closures to their creation contexts; they are always created in the global scope.</p>
+
+  <p>When running them, they will only be able to access their own local variables and global ones, not the ones from the scope in which the <code>AsyncGeneratorFunction</code> constructor was called.</p>
+
+  <p>This is different from using {{jsxref("Global_Objects/eval", "eval")}} with code for a generator function expression.</p>
+</div>
+
+<p>Invoking the <code>AsyncGeneratorFunction</code> constructor as a function (without using
+  the <code>new</code> operator) has the same effect as invoking it as a constructor.</p>
+
+<h2 id="Examples">Examples</h2>
+
+<h3 id="using_the_constructor">Using the constructor</h3>
+
+<p>The following example uses the <code>AsyncGeneratorFunction</code> constructor to create an async generator function.</p>
+
+<pre class="brush: js">const AsyncGeneratorFunction = Object.getPrototypeOf(async function*(){}).constructor;
+const asyncGen = new AsyncGeneratorFunction('a', 'yield a * 2');
+const asyncIterator = asyncGen(10);
+console.log((await asyncIterator.next()).value); // 20
```

Comment:
I had not considered that top-level await could be problematic for examples due to browser compatibility (currently being behind a flag in Firefox nightly and all). Should we update this one here too?

```suggestion
asyncIterator.next().then(res => console.log(res.value)); // 20
```